### PR TITLE
Add webm support

### PR DESCRIPTION
**Pull request by CEBATOR**
### Background
WEBM support was down for a while, and it's frustrating given that most video content on 2ch is in that exact format. Since iOS has no native support for that format, I had to use [VLCKit](https://code.videolan.org/videolan/VLCKit)

### What has been done
1. Updated target to deploy target to iOS 9 (needed 9.0 for safe area support.)
2. Add VLCKit pod 
3. Integrated  Swift into the project
4. Created VLCPlayer view controller
5. Add Icons for the player view
6. Update readme file to reflect the new minimal iOS version

### How to test
0. Proofread the code (might have grammatical errors)
1. Start the app
2. Open a thread that has WebM files
3. Tap on any WebM file
4. Make sure the video starts muted
5. Test that the controls are working properly
6. Make sure that the play/pause and mute/unmute buttons have opposite action relative to the player state. (ex. should be pause while video playing)

### TODO (sometimes in the future)
- Player controls have a bit of delay on tap
- There is a bug that will not let the video to play from the start once the playback is finished.
- Update deprecated code.
- Player controls are not hiding.
- No error is presented when the URL to WebM is broken

### CODE REVIEW

**Comment by BadMachine** [comment]
@CEBATOR Why this app is not available in Russian store?

**Comment by CEBATOR** [comment]
@BadMachine A few reasons
* it has a lot of deprecated code, that needs to be updated before it can be submitted
* even though the NSFW boards are hidden behind a pop-up, the possibility to add them could be enough for Apple to not approve the app

I might try to upload it to appstore after I finish updating, given that GLP3 license allows that.

In the meantime, you can build the app yourself, if you have access to a mac.

**Comment by BadMachine** [comment]
@CEBATOR Thanks for your answer!

File: Podfile [hunk 1]
```diff
@@ -14,6 +14,7 @@ abstract_target 'BasePods' do
 	pod 'Texture', '2.5.1'
 	pod 'TUSafariActivity', '1.0.4'
 	pod 'YapDatabase', '3.0.2'
+	pod 'MobileVLCKit', '~>3.3.0'
```
Comment (8of) [inline Podfile:17]: Let's set the exact version here (without `~>`) for stability, for now.

File: dvach-browser.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1175,17 +1206,21 @@
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				DEFINES_MODULE = YES;
+				DEVELOPMENT_TEAM = Q9324QX3V7;
```
Comment (8of) [inline dvach-browser.xcodeproj/project.pbxproj:1210]: Auto-change, let's remove it.

File: dvach-browser/VLCPlayerViewController.swift [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+//  dvach-browser
+//
+//  Created by Victor Cebanu on 2/3/20.
+//  Copyright © 2020 8of. All rights reserved.
+//
+
+import Foundation
+
+final class VLCPlayerViewController: UIViewController, VLCPlayerViewInput {
+    var interactor: VLCPlayerViewOutput!
```
Comment (8of) [inline dvach-browser/VLCPlayerViewController.swift:12]: Let's use dependency injection.
Make our own `init` and set `interactor` inside.

File: dvach-browser/VLCPlayerViewController.swift [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    var player: VLCMediaPlayer?
+
+    @IBOutlet weak var playerView: UIView!
```
Comment (8of) [inline dvach-browser/VLCPlayerViewController.swift:15]: I suggest we mark everything `private` if we aren't accessing it from outside. Even `IBOutlet`s and `IBAction`s.

File: dvach-browser/VLCPlayerViewController.swift [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupPlayer()
+        interactor.viewDidLoad()
+    }
+
+    // IBAction
+
+    @IBAction func progressBarWasMoved(_ sender: UISlider) {
```
Comment (8of) [inline dvach-browser/VLCPlayerViewController.swift:33]: We can move every private method to a private local extension. Code will be more organized, robust and predictable.

File: dvach-browser/Views/ViewGenerators/DVBAlertGenerator.m [hunk 6]
```diff
@@ -36,18 +36,6 @@ + (UIAlertController *)ageCheckAlert {
   return alertController;
 }
 
-+ (UIAlertController *)webmDeprecatedAlert {
-    UIAlertController *alertController = [UIAlertController
-                                          alertControllerWithTitle:NSLS(@"DEPRECATED")
```
Comment (8of) [inline dvach-browser/Views/ViewGenerators/DVBAlertGenerator.m:41]: We can get rid of "Deprecated" in both localizations as well.

**Review by 8of — COMMENTED** [review]
- Good PR. Extra ❤️ for using final and private in the right places.
- Make suggested small improvement and I'll merge it.
- About "a bug that will not let the video to play from the start once the playback is finished". Is there a simple option in VLC to loop the video? It would solve the problem for now.
- While we are at it, add credentials to the Readme at the end. Something like `WebM support by [your nick/name and link]` will do.

File: dvach-browser/VLCPlayerViewController.swift (hunk 3 above)
Comment (CEBATOR) [reply dvach-browser/VLCPlayerViewController.swift:12]: we inject the view controller into the interactor
we can't do it both ways
check `VLCPlayerScreenBuilder.swift`

**Comment by 8of** [comment]
Surprisingly, several people asked me lately to finally merge this PR or add WebM support at least somehow.

It was an Objective C-only project so far. I decided to leave it that way and ported the code to Objective C.

Current video controls implementation is really frustrating. Because of delay after tap. The only control I left is a tap on video view to mute / unmute current video.

I'll try to look into the problem later if I'll have time. Maybe, I'll implement full thread playlist for convenient webm threads surfing.

Anyway, you pushed the project in the right direction and new credit in README file is well deserved. Cheers! 🙌

**Pull request closed without merging**